README.md

```diff
@@ -45,102 +53,241 @@ console.log(result);
+
+## Kinds Of Errors
+Broadly speaking, errors come in two kinds [[8]](#resources):
+* **Programmer errors**: These are _bugs_; they are unintended and/or unanticipated behaviour of the code, and they can only be fixed by changing the code (e.g. calling a function with the wrong number of arguments)
+* **Operational errors**: These are runtime errors that are usually caused by some external factor (e.g. any kind of network error, failure to read a file, running out of memory, etc.)
+
+How you should handle any given error depends on what kind of error it is. Operational errors are a normal part of the issues a program must deal with. They typically should not cause the program to terminate or behave unexpectedly. By contrast, programmer errors are by definition unanticipated, and may potentially leave the application with unpredictable state and behaviour. In this case it is usually best to terminate the program.
+
+There is, however, no blanket rule for what to do; each error represents a specific problem in the context of an entire application and the appropriate response to it will be heavily context dependent.
+
+## Approaches
+Good error handling is typically not something that can just be bolted onto an existing program as an afterthought. Well conceived error handling will affect the structure of the code. In JavaScript and Node.js, there are a number of approaches, some of which are explored below.
+
+### General Principles
+Regardless of the chosen approach, there are some principles which can be generally applied:
+1. **Be consistent, not ad-hoc.**
+  * Inconsistent approaches to error handling will complicate your code and make it much harder to reason about.
+2. **Try to trip into a failure code path as early as possible.**
+  * For example, it may be tempting to return default values in the case of an error and allow the application to continue as normal.
+  * This may be appropriate in some cases, but can often cover up the root cause of an error and make it difficult to track down, or result in unhelpful error messages.
+3. **Propagate errors to a part of the application that has sufficient context to know how to deal with them.**
+  * Many times, we will write generic functions to perform common actions, like making a network request.
+  * If the network request fails, the generic function cannot infer the appropriate response, because it doesn't know which part of the application it has been called from.
+  * It should therefore try to propagate the error to its caller instead of trying to recover directly.
+
+### Illustrative Example
+To illustrate the three approaches we will cover, we will use the same simple example in each, so that comparisons are easier. Imagine you intend to write a function `applyToInteger`, with the following signature:
+```js
+applyToInteger(func: Function, integer: Number)
```

Comment:
Reading over your contributions to this workshop I'm impressed by the depth you've added to it. 

I'm unconfident to maintain/contribute further to this as I see this workshop (and the subject of error handling in JS) grow *very* rapidly (for all good reasons) and have much higher precedence in the future- perhaps even split into several workshops...

Anyway...
```
applyToInteger(func: Function, integer: Number)
```
I think you should add an aside point here about TypeScript/Flow & languages that extend JS with static type checking. Perhaps just with links for further reading?